csharp/test/Apache.Arrow.Tests/ArrayBuilderTests.cs

```diff
@@ -38,6 +39,101 @@ public void PrimitiveArrayBuildersProduceExpectedArray()
             TestArrayBuilder<DoubleArray, DoubleArray.Builder>(x => x.Append(10).Append(20).Append(30));
         }
 
+
+        [Fact]
+        public void ListArrayBuilder()
+        {
+            var list = CreateList();
+
+            Assert.Equal(
+                new List<string> { "1" },
+                ConvertStringArrayToList(list.GetSlicedValues(0) as StringArray));
+            Assert.Equal(
+                new List<string> { "22", "33" },
+                ConvertStringArrayToList(list.GetSlicedValues(1) as StringArray));
+            Assert.Equal(
+                new List<string> { "444", "555", "666" },
+                ConvertStringArrayToList(list.GetSlicedValues(2) as StringArray));
+
+            ListArray CreateList()
```

Comment:
Why separate this into a nested method that is only called once? Why not just put the below code inline in the `ListArrayBuilder()` method?

Reply:
I intended to improve readability by separating the process of creating test data from the process of checking.
Of course, there is no need to separate them in term of functionality.
If you feel this separation does not makes much sense, I'll modify it.

Reply:
This isn't a super important comment to address - it was more of a question. I just find it a little harder to read since the nested method kind of "gets in the way" (in my opinion).

```C#
            var list = CreateList();

            Assert.Equal(
                new List<string> { "1" },
                ConvertStringArrayToList(list.GetSlicedValues(0) as StringArray));
            Assert.Equal(
                new List<string> { "22", "33" },
                ConvertStringArrayToList(list.GetSlicedValues(1) as StringArray));
            Assert.Equal(
                new List<string> { "444", "555", "666" },
                ConvertStringArrayToList(list.GetSlicedValues(2) as StringArray));

            ListArray CreateList()
            {
                var stringField = new Field("item", StringType.Default, true);
                var listBuilder = new ListArray.Builder(new ListType(stringField));
                var valueBuilder = listBuilder.ValueBuilder as StringArray.Builder;
                Assert.NotNull(valueBuilder);
                listBuilder.Append();
                valueBuilder.Append("1");
                listBuilder.Append();
                valueBuilder.Append("22").Append("33");
                listBuilder.Append();
                valueBuilder.Append("444").Append("555").Append("666");

                return listBuilder.Build();
            }
```

vs.

```C#
            var stringField = new Field("item", StringType.Default, true);
            var listBuilder = new ListArray.Builder(new ListType(stringField));
            var valueBuilder = listBuilder.ValueBuilder as StringArray.Builder;
            Assert.NotNull(valueBuilder);
            listBuilder.Append();
            valueBuilder.Append("1");
            listBuilder.Append();
            valueBuilder.Append("22").Append("33");
            listBuilder.Append();
            valueBuilder.Append("444").Append("555").Append("666");

            var list = listBuilder.Build();

            Assert.Equal(
                new List<string> { "1" },
                ConvertStringArrayToList(list.GetSlicedValues(0) as StringArray));
            Assert.Equal(
                new List<string> { "22", "33" },
                ConvertStringArrayToList(list.GetSlicedValues(1) as StringArray));
            Assert.Equal(
                new List<string> { "444", "555", "666" },
                ConvertStringArrayToList(list.GetSlicedValues(2) as StringArray));
```

I just find that the latter reads more straight-forward (top to bottom).

Reply:
Thank you for your feedback.
I myself had wondered whether this approach is easy to read for others.
Hearing your opinion, I decided to avoid using this inner method.